src/main/java/edu/hm/hafner/analysis/parser/IssuesFilter.java

```diff
@@ -0,0 +1,188 @@
+        return this;
+    }
+
+    public IssuesFilter includePriority(String priorityRegEx){
+        includeFilterSet = true;
+        patternIncludePriority = Pattern.compile(priorityRegEx);
+        return this;
+    }
+
+    public IssuesFilter includeMessage(String messageRegEx){
+        includeFilterSet = true;
+        patternIncludeMessage = Pattern.compile(messageRegEx);
+        return this;
+    }
+
+    public IssuesFilter includeDescription (String descriptionRegEx){
+        includeFilterSet = true;
+        patternIncludeDescription = Pattern.compile(descriptionRegEx);
+        return this;
+    }
+
+    public IssuesFilter includeType (String typeRegEx){
+        includeFilterSet = true;
+        patternIncludeType = Pattern.compile(typeRegEx);
+        return this;
+    }
+
+    public IssuesFilter includeColumnStart (String columnStartRegEx){
+        includeFilterSet = true;
+        patternIncludeColumnStart = Pattern.compile(columnStartRegEx);
+        return this;
+    }
+
+    public IssuesFilter includeColumnEnd (String columnEndRegEx){
+        includeFilterSet = true;
+        patternIncludeColumnStart = Pattern.compile(columnEndRegEx);
+        return this;
+    }
+
+    public IssuesFilter excludeFilename(String filenameRegEx){
+        excludeFilterSet = true;
+        patternExcludeFilename = Pattern.compile(filenameRegEx);
+        return this;
+    }
+
+    public IssuesFilter excludeCategory(String categoryRegEx){
+        excludeFilterSet = true;
+        patternExcludeCategory = Pattern.compile(categoryRegEx);
+        return this;
+    }
+
+    public IssuesFilter excludePriority(String priorityRegEx){
+        excludeFilterSet = true;
+        patternExcludePriority = Pattern.compile(priorityRegEx);
+        return this;
+    }
+
+    public IssuesFilter excludeMessage(String messageRegEx){
+        excludeFilterSet = true;
+        patternExcludeMessage = Pattern.compile(messageRegEx);
+        return this;
+    }
+
+    public IssuesFilter excludeDescription (String descriptionRegEx){
+        excludeFilterSet = true;
+        patternExcludeDescription = Pattern.compile(descriptionRegEx);
+        return this;
+    }
+
+    public IssuesFilter excludeType (String typeRegEx){
+        excludeFilterSet = true;
+        patternExcludeType = Pattern.compile(typeRegEx);
+        return this;
+    }
+
+    public IssuesFilter excludeColumnStart (String columnStartRegEx){
+        excludeFilterSet = true;
+        patternExcludeColumnStart = Pattern.compile(columnStartRegEx);
+        return this;
+    }
+
+    public IssuesFilter excludeColumnEnd (String columnEndRegEx){
+        excludeFilterSet = true;
+        patternExcludeColumnStart = Pattern.compile(columnEndRegEx);
+        return this;
+    }
+
+    public Issues filter(){
+        Issues includedIssues = new Issues();
+        Issues completeFilteredIssues = new Issues();
+
+        if (includeFilterSet == true) {
```

Comment:
Es werden immer alle Pattern durchlaufen, auch wenn diese gar nicht benötigt werden. Performanter wär, nur die Pattern zu benutzen die auch konfiguriert wurden. 
Wenn du einen Builder hättest, könntest du die Aufbereitung und Kombination der einzelnen Pattern einmalig in der build-Methode anwenden, so das der Filter bei öfterer Anwendung dies nicht mehr tun muss.